tensordict/prototype/tensorclass.py

```diff
@@ -812,18 +680,11 @@ def _gather(self, dim: int, index: torch.Tensor, out: TensorDictBase | None = No
     if dim < 0:
         dim = self.batch_dims + dim
 
-    def _get_out(key, nontensor):
-        if out is None:
-            return None
-        if nontensor:
-            return out._non_tensordict[key]
-        else:
-            return out._tensordict
-
-    non_tensor = copy(self._non_tensordict)
-    return self._from_tensordict(
-        self._tensordict.gather(dim=dim, index=index, out=_get_out(None, False)),
-        non_tensor,
+    return _from_tensordict_with_copy(
+        self,
+        self._tensordict.gather(
+            dim=dim, index=index, out=out._tensordict if out is not None else None
```

Comment:
I wanted to drop `_gather` but I can't figure out how to generically intercept `out` and replace it with `out._tensordict`.